app/merginapi.cpp

```diff
@@ -1196,88 +1168,100 @@ void MerginApi::updateInfoReplyFinished()
 
-    QList<MerginFile> filesToDownload;
-    qint64 totalSize = 0;
-    for ( QString filePath : transaction.diff.remoteAdded )
-    {
-      MerginFile file = serverProject.fileInfo( filePath );
-      file.chunks = generateChunkIdsForSize( file.size ); // doesnt really matter whats there, only how many chunks are expected
-      filesToDownload << file;
-      totalSize += file.size;
-    }
+  LocalProjectInfo projectInfo = mLocalProjects.projectFromMerginName( projectFullName );
+  if ( projectInfo.isValid() )
+  {
+    transaction.projectDir = projectInfo.projectDir;
+  }
+  else
+  {
+    QString projectNamespace;
+    QString projectName;
+    extractProjectName( projectFullName, projectNamespace, projectName );
 
-    for ( QString filePath : transaction.diff.remoteUpdated )
-    {
-      MerginFile file = serverProject.fileInfo( filePath );
-      file.chunks = generateChunkIdsForSize( file.size ); // doesnt really matter whats there, only how many chunks are expected
-      filesToDownload << file;
-      totalSize += file.size;
-    }
+    // project has not been downloaded yet - we need to create a directory for it
+    transaction.projectDir = createUniqueProjectDirectory( projectName );
+    transaction.firstTimeDownload = true;
+  }
 
-    // also download files which were changed both on the server and locally (the local version will be renamed as conflicting copy)
-    for ( QString filePath : transaction.diff.conflictRemoteUpdatedLocalUpdated )
-    {
-      MerginFile file = serverProject.fileInfo( filePath );
-      file.chunks = generateChunkIdsForSize( file.size ); // doesnt really matter whats there, only how many chunks are expected
-      filesToDownload << file;
-      totalSize += file.size;
-    }
+  Q_ASSERT( !transaction.projectDir.isEmpty() );  // that would mean we do not have entry -> fail getting local files
 
-    // also download files which were added both on the server and locally (the local version will be renamed as conflicting copy)
-    for ( QString filePath : transaction.diff.conflictRemoteAddedLocalAdded )
-    {
-      MerginFile file = serverProject.fileInfo( filePath );
-      file.chunks = generateChunkIdsForSize( file.size ); // doesnt really matter whats there, only how many chunks are expected
-      filesToDownload << file;
-      totalSize += file.size;
-    }
+  QList<MerginFile> localFiles = getLocalProjectFiles( transaction.projectDir + "/" );
+  MerginProjectMetadata serverProject = MerginProjectMetadata::fromJson( data );
+  MerginProjectMetadata oldServerProject = MerginProjectMetadata::fromCachedJson( transaction.projectDir + "/" + sMetadataFile );
 
-    transaction.totalSize = totalSize;
-    transaction.files = filesToDownload;
+  transaction.projectMetadata = data;
+  transaction.version = serverProject.version;
+  transaction.diff = compareProjectFiles( oldServerProject.files, serverProject.files, localFiles );
+  InputUtils::log( "update", transaction.diff.dump() );
```

Comment:
Any particular reason why log "update " instead url as always?

Reply:
we don't have the URL available here (and it's not worth having another argument just for that). I would like to revisit these log contexts later... Over time I would like to see fewer URLs in the log - e.g. only when request is started or reply has finished. Normally we can simply just use project full name for context...